vmprof/__init__.py

```diff
@@ -47,3 +49,28 @@ def enable_jitlog(fileno):
             be broken.
         """
         _vmprof.enable_jitlog(fileno)
+
+
+@contextlib.contextmanager
+def profile(outfile, pipecmd=None, period=DEFAULT_PERIOD, memory=False):
+    """Utility context manager which calls vmprof.enable() and vmprof.disable().
+
+    outfile is path to output file.
+
+    This function support compression via pipe command::
+
+        with vmprof.profile("out.prof.gz", pipecmd=["/usr/bin/gzip", "-4"]):
+            main()
+    """
+    with open(outfile, 'wb') as of:
+        proc = None
+        fileno = of.fileno()
+        if pipecmd is not None:
+            proc = subprocess.Popen(pipecmd, bufsize=-1, stdin=subprocess.PIPE, stdout=of.fileno())
+            fileno = proc.stdin.fileno()
+        enable(fileno, period, memory)
+        yield
```

Comment:
I believe you have to use

```py
try:
    yield
finally:
    disable()
    ...
```

to account for the case that an exception is thrown in the profiled function.